tracer/src/Datadog.Trace/LifetimeManager.cs

```diff
@@ -80,6 +80,12 @@ private void CurrentDomain_ProcessExit(object? sender, EventArgs e)
 
         private void CurrentDomain_UnhandledException(object? sender, UnhandledExceptionEventArgs e)
         {
+            if (e == null || e.ExceptionObject is OutOfMemoryException)
```

Comment:
While we're doing this, are there other exception types we should include here in addition to `OutOfMemoryException`?

Reply:
Maybe StackOverflowException but I don't expect this method to be called at all when it happens 🤔  All the other exceptions are fine